src/Compilers/CSharp/Portable/Binder/Binder_Statements.cs

```diff
@@ -1668,26 +1668,32 @@ private DiagnosticInfo GetBadEventUsageDiagnosticInfo(EventSymbol eventSymbol)
                 new CSDiagnosticInfo(ErrorCode.ERR_BadEventUsageNoField, leastOverridden);
         }
 
-        internal static bool AccessingAutoPropertyFromConstructor(BoundPropertyAccess propertyAccess, Symbol fromMember)
+        internal static bool IsPropertyAssignedThroughBackingField(BoundPropertyAccess propertyAccess, Symbol fromMember)
         {
-            return AccessingAutoPropertyFromConstructor(propertyAccess.ReceiverOpt, propertyAccess.PropertySymbol, fromMember);
+            return IsPropertyAssignedThroughBackingField(propertyAccess.ReceiverOpt, propertyAccess.PropertySymbol, fromMember);
         }
 
-        private static bool AccessingAutoPropertyFromConstructor(BoundExpression receiver, PropertySymbol propertySymbol, Symbol fromMember)
+        /// <summary>
+        /// Determines whether a property is assigned through the backing field within a symbol <paramref name="fromMember"/>
+        /// </summary>
+        /// <remarks>
+        /// A property is assigned through backing field within a constructor or a field assignment.
+        /// </remarks>
+        private static bool IsPropertyAssignedThroughBackingField(BoundExpression receiver, PropertySymbol propertySymbol, Symbol fromMember)
```

Comment:
There is a minor behavioral change here if the property doesn't have any accessors. Previously this will return `false` due to the "WithGetAccessor" check, but now it returns true.

The behavior, I think, doesn't matter that much. We now get less diagnostics.

The diagnostics (for a property without accessors and an attempt to write to it in constructor) were:

```
error CS0548: 'C.P': property or indexer must have at least one accessor
error CS0200: Property or indexer 'C.P' cannot be assigned to -- it is read only
```

But now only CS0548 is produced. I don't have a strong opinion about which behavior it should be. Implementation-wise, the code feels more natural without the get accessor check.

Reply:
I'll retain the old behavior for now since this is meant to be a refactoring only, so no behavioral changes should be observed.